src/crypto/CHIPCryptoPAL.h

```diff
@@ -1285,5 +1285,73 @@ enum class MatterOid
  **/
 CHIP_ERROR ExtractDNAttributeFromX509Cert(MatterOid matterOid, const ByteSpan & certificate, uint16_t & id);
 
+/**
+ * @brief Opaque context used to protect the symmetric key. The key operations must
+ *        be performed without exposing the protected key value.
+ */
+class SymmetricKeyContext
+{
+public:
+    virtual ~SymmetricKeyContext() = default;
+    /**
+     * @brief Perform the message encryption as described in 4.7.2. (Security Processing of Outgoing Messages)
+     * @param[inout] plaintext     Outgoing message payload.
+     * @param[in] aad           Additional data (message header contents)
+     * @param[in] nonce         Nonce (Security Flags | Message Counter | Source Node ID)
+     * @param[out] out_mic      Outgoing Message Integrity Check
+     * @return CHIP_ERROR
+     */
+    virtual CHIP_ERROR EncryptMessage(MutableByteSpan & plaintext, const ByteSpan & aad, const ByteSpan & nonce,
+                                      MutableByteSpan & out_mic) const = 0;
+    /**
+     * @brief Perform the message decryption as described in 4.7.3.(Security Processing of Incoming Messages)
+     * @param ciphertext[inout] Incoming encrypted payload
+     * @param aad[in]   Additional data (message header contents)
+     * @param nonce[in] Nonce (Security Flags | Message Counter | Source Node ID)
+     * @param mic[in]   Incoming Message Integrity Check
+     * @return CHIP_ERROR
+     */
+    virtual CHIP_ERROR DecryptMessage(MutableByteSpan & ciphertext, const ByteSpan & aad, const ByteSpan & nonce,
+                                      const ByteSpan & mic) const = 0;
+
+    /**
+     * @brief Perform privacy encoding as described in 4.8.2. (Privacy Processing of Outgoing Messages)
+     * @param header[in/out]    Message header to encrypt
+     * @param session_id[in]    Outgoing SessionID
+     * @param payload[in]       Encrypted payload
+     * @param mic[in]       Outgoing Message Integrity Check
+     * @return CHIP_ERROR
+     */
+    virtual CHIP_ERROR EncryptPrivacy(MutableByteSpan & plaintext, const ByteSpan & nonce) const = 0;
+
+    /**
+     * @brief Perform privacy decoding as described in 4.8.3. (Privacy Processing of Incoming Messages)
+     * @param header[in/out]    Message header to decrypt
+     * @param session_id[in]    Incoming SessionID
+     * @param payload[in]       Encrypted payload
+     * @param mic[in]           Outgoing Message Integrity Check
```

Comment:
These params are absent. Please review docstrings for all new methods here...

Reply:
I have fixed the signature of the methods.